src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -570,39 +569,49 @@ else if (quizExercise.isSubmissionAllowed()) {
      * @param participationId the participationId of the participation to delete
      * @param deleteBuildPlan True, if the build plan should also get deleted
      * @param deleteRepository True, if the repository should also get deleted
-     * @param principal The identity of the user accessing this resource
      * @return the ResponseEntity with status 200 (OK) or 403 (FORBIDDEN)
      */
-    @DeleteMapping("/guided-tour/participations/{participationId}")
+    @DeleteMapping("guided-tour/participations/{participationId}")
     @PreAuthorize("hasRole('USER')")
     public ResponseEntity<Void> deleteParticipationForGuidedTour(@PathVariable Long participationId, @RequestParam(defaultValue = "false") boolean deleteBuildPlan,
-            @RequestParam(defaultValue = "false") boolean deleteRepository, Principal principal) {
+            @RequestParam(defaultValue = "false") boolean deleteRepository) {
         StudentParticipation participation = studentParticipationRepository.findByIdElseThrow(participationId);
 
         if (participation instanceof ProgrammingExerciseParticipation && !featureToggleService.isFeatureEnabled(Feature.PROGRAMMING_EXERCISES)) {
-            return forbidden();
+            throw new AccessForbiddenException("Not allowed");
         }
 
         User user = userRepository.getUserWithGroupsAndAuthorities();
 
         // Allow all users to delete their own StudentParticipations if it's for a tutorial
         if (participation.isOwnedBy(user)) {
             checkAccessPermissionAtLeastStudent(participation, user);
-            if (!guidedTourConfiguration.isExerciseForTutorial(participation.getExercise())) {
-                return forbidden();
-            }
+            guidedTourConfiguration.checkExerciseForTutorialElseThrow(participation.getExercise());
         }
         else {
-            return forbidden();
+            throw new AccessForbiddenException("Not allowed");
```

Comment:
I would like more details in the exceptions here; that makes debugging much easier.